src/views/eip4626/EIP4626.tsx

```diff
@@ -4,17 +4,39 @@ import Admin from "@/components/eip4626/Admin";
 import User from "@/components/eip4626/User";
 import NoWalletConnected from "@/components/NoWalletConnected";
 import { WatchContractEventReturnType } from "@/services/contracts/watchContractEvent";
-import { useContext, useEffect } from "react";
+import { useContext, useEffect, useState } from "react";
 import { Eip4626Context } from "@/contexts/Eip4626Context";
 import {
+  readTokenName,
+  readHtsTokenTokenAddress,
   watchHtsTokenFactoryTokenDeployedEvent,
   watchVaultFactoryVaultDeployedEvent,
 } from "@/services/contracts/wagmiGenActions";
 
 export default function EIP4626() {
   const { accountId } = useWalletInterface();
-  const { setDeployedVaults, setDeployedProxyHtsTokens } =
-    useContext(Eip4626Context);
+  const [tabIndex, setTabIndex] = useState(0);
+  const {
+    setDeployedVaults,
+    setDeployedProxyHtsTokens,
+    setDeployedHtsTokenNames,
+    deployedProxyHtsTokens,
+  } = useContext(Eip4626Context);
+
+  useEffect(() => {
+    deployedProxyHtsTokens.forEach((token) => {
+      readHtsTokenTokenAddress({}, token).then((value) => {
+        readTokenName({}, (value as unknown as `0x${string}`[])[0]).then(
+          (data) => {
+            setDeployedHtsTokenNames((prev: any) => ({
+              ...prev,
+              [token]: data[0],
+            }));
+          },
+        );
+      });
+    });
+  }, [deployedProxyHtsTokens, setDeployedHtsTokenNames]);
```

Comment:
I wonder if we launch the chain of requests every time whenever deployedProxyHtsTokens gets populated with new batch of tokens from the chain.
I imagine refactoring these into dependant queries would help us prevent unnecessary requests.

Reply:
> I wonder if we launch the chain of requests every time whenever deployedProxyHtsTokens gets populated with new batch of tokens from the chain. I imagine refactoring these into dependant queries would help us prevent unnecessary requests.

Agree, i opened a new issue as it requires some more accurate work to be done:
https://github.com/web3-nomad/accelerator-defi-EIP-ui/issues/78